webrtc.html

```diff
@@ -8335,6 +8349,45 @@ <h3><dfn>RTCIceTransport</dfn> Interface</h3>
           <code><a>RTCPeerConnection</a></code> object associated with this
           <a>ICE Agent</a>.</p>
         </li>
+        <li>
+          <p>If <var>connection</var>'s <a>[[\IsClosed]]</a> slot is
+          <code>true</code>, abort these steps.</p>
+        </li>
+        <li>
+          <p>If either <var>connection</var>.<a>[[\PendingLocalDescription]]</a>
+          or <var>connection</var>.<a>[[\CurrentLocalDescription]]</a> are not
+          <code>null</code>, and represent the ICE <a>generation</a> for which
+          <var>candidate</var> was gathered,
+          <a href="#dfn-surface-a-candidate">surface the candidate</a> with
+          <var>candidate</var> and <var>connection</var>, and abort these steps.
+          </p>
+        </li>
+        <li>
+          <p>Otherwise, append <var>candidate</var> to
+          <var>connection</var>.<a>[[\EarlyCandidates]]</a>.</p>
+        </li>
+      </ol>
+      <p>To <dfn>release early candidates</dfn> of a <var>connection</var>,
+      run the following steps:</p>
+      <ol>
+        <li>
+          <p>For each candidate, <var>candidate</var>, in
+          <var>connection</var>.<a>[[\EarlyCandidates]]</a>, if
+          either <var>connection</var>.<a>[[\PendingLocalDescription]]</a> or
+          <var>connection</var>.<a>[[\CurrentLocalDescription]]</a> are not
+          <code>null</code>, and represent the ICE <a>generation</a>
+          of <var>candidate</var>, then queue a task to <a
+            href="#dfn-surface-a-candidate">surface the candidate</a>
+          with <var>candidate</var> and <var>connection</var>.</p>
```

Comment:
An early candidate is one we haven't fired an event for yet, so why are we filtering out candidates here? Is this in case of an ICE restart where they have become invalid already?

Otherwise - speaking of ICE restart: should a successful ICE restart cause [[EarlyCandidates]] to be cleared?

If this is something we always do in stable state - should we perhaps only look at [[CurrentLocalDescription]], or am I missing something?

Reply:
> why are we filtering out candidates here?

Some prose contortions led me to think this was needed. I think you're right it's not.

> speaking of ICE restart: should a successful ICE restart cause [[EarlyCandidates]] to be cleared?

Wouldn't [[EarlyCandidates]] already be emtpy? How can a restart be "successful" without SLD() having succeeded?

> If this is something we always do in stable state - should we perhaps only look at [[CurrentLocalDescription]], or am I missing something?

You're right. I'll remove the filtering outright.